src/tree.rs

```diff
@@ -11,6 +11,8 @@
 // under the License.
 
 use std::cmp::Ordering;
+use std::collections::HashSet;
```

Comment:
Ok, I was under the impression that hashbrown's `HashSet` is now the default `HashSet`.

From the [hashbrown README](https://github.com/rust-lang/hashbrown):
> Since Rust 1.36, this is now the HashMap implementation for the Rust standard library. However you may still want to use this crate instead since it works in environments without std, such as embedded systems and kernels.

Reply:
It is the hashbrown crate is the upstream for the std lib hashset and hashmap. For our purposes what the crate does that's different from the std version is use ahash as the default hash algorithm and also has optional support for rayon (it also has additional options like nostd, etc which we don't leverage here).

Reply:
Ok, that makes sense -- thanks for explaining.